components/support/nimbus-fml/src/client/mod.rs

```diff
@@ -320,3 +320,294 @@ mod unit_tests {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod string_aliases {
+    use super::{test_helper::client, *};
+
+    #[test]
+    fn test_simple_feature() -> Result<()> {
+        let client = client("string-aliases.fml.yaml", "storms")?;
+        let inspector = {
+            let i = client.get_feature_inspector("my-simple-team".to_string());
+            assert!(i.is_some());
+            i.unwrap()
+        };
+
+        // -> feature my-sports:
+        //      player-availability: Map<PlayerName, Boolean> (PlayerName is the set of strings in this list)
+        //      captain: Option<PlayerName>
+        //      the-team: List<SportName> (SportName is the set of string that are keys in this map)
+
+        // Happy path. This configuration is internally consistent.
+        let errors = inspector.get_errors(
+            r#"{
+                "captain": "Babet",
+                "the-team": ["Babet", "Elin", "Isha"]
+            }"#
+            .to_string(),
+        );
+        assert_eq!(None, errors);
+
+        // ----------------------------
+        // Donkey cannot be the captain
+        // Donkey is not a key in the default) player-availability map.
```

Comment:
Oh I didn't know we do this pattern here where we put multiple test cases inside a single test.  I usually like to split this up into individual tests so if one fails it's clearer exactly which case failed rather than having to look into the test to know which specific case failed.  But don't mind me, just a drive by observation.